lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -178,40 +186,24 @@ def index(
             deprecated=True,  # Marked as deprecated as it seems just like '/api/histories/deleted'
         ),
     ) -> List[AnyHistoryView]:
-        return self.service.index(
-            trans, serialization_params, filter_query_params, deleted_only=deleted, all_histories=all
-        )
-
-    @router.get(
-        "/api/histories/query",
-        summary="Returns histories available to the current user.",
-    )
-    def query(
-        self,
-        response: Response,
-        trans: ProvidesUserContext = DependsOnTrans,
-        limit: Optional[int] = LimitQueryParam,
-        offset: Optional[int] = OffsetQueryParam,
-        show_own: bool = ShowOwnQueryParam,
-        show_published: bool = ShowPublishedQueryParam,
-        show_shared: bool = ShowSharedQueryParam,
-        sort_by: HistorySortByEnum = SortByQueryParam,
-        sort_desc: bool = SortDescQueryParam,
-        search: Optional[str] = SearchQueryParam,
-    ) -> HistoryQueryResultList:
-        payload = HistoryIndexQueryPayload.construct(
-            show_own=show_own,
-            show_published=show_published,
-            show_shared=show_shared,
-            sort_by=sort_by,
-            sort_desc=sort_desc,
-            limit=limit,
-            offset=offset,
-            search=search,
-        )
-        entries, total_matches = self.service.index_query(trans, payload, include_total_count=True)
-        response.headers["total_matches"] = str(total_matches)
-        return entries
+        if search is None:
+            return self.service.index(
+                trans, serialization_params, filter_query_params, deleted_only=deleted, all_histories=all
+            )
+        else:
+            payload = HistoryIndexQueryPayload.construct(
```

Comment:
I don't love this, I would have preferred an "adapter" step to try to convert those new query parameters to fit in the `filter_query_params` instead of the complete branching. Although I haven't checked if there is any limitation on this.
Anyway, I guess is better than having the extra route for now.

Reply:
I agree with you, this is not very pretty. I think we have to decide wether to combine the two code paths or deprecate the legacy path moving forward. I would favor the latter, since this would also streamline the client request, which in the history panel, converts the modern search string to q/qv values as of right now.

Reply:
If we go ahead with deprecating q/qv, it would be nice if the new syntax still supported queries that we currently expose in BioBlend, e.g.:

https://github.com/galaxyproject/bioblend/blob/main/bioblend/galaxy/histories/__init__.py#L135

Not saying that it doesn't, I just wanted to flag it.
Then BioBlend can transparently use one syntax or the other, depending on the Galaxy version.